src/shogun/kernel/LogKernel.h

```diff
@@ -56,47 +56,23 @@ class LogKernel: public Kernel
 	 */
 	virtual bool init(std::shared_ptr<Features> l, std::shared_ptr<Features> r);
 
-	/**
-	 * @return kernel type
-	 */
 	virtual EKernelType get_kernel_type() { return K_POWER; }
 
-	/**
-	 * @return type of features
-	 */
 	virtual EFeatureType get_feature_type() { return m_distance->get_feature_type(); }
 
-	/**
-	 * @return class of features
-	 */
 	virtual EFeatureClass get_feature_class() { return m_distance->get_feature_class(); }
 
-	/**
-	 * @return name of kernel
-	 */
 	virtual const char* get_name() const { return "LogKernel"; }
 
 	virtual ~LogKernel();
 
 protected:
-	/**
-	 * compute kernel for specific feature vectors
-	 * corresponding to [idx_a] of left-side and [idx_b] of right-side
-	 * @param idx_a left-side index
-	 * @param idx_b right-side index
-	 * @return kernel value
-	 */
 	virtual float64_t compute(int32_t idx_a, int32_t idx_b);
 
-private:
-	void init();
-
 protected:
-	/// distance to be used
-	std::shared_ptr<Distance> m_distance;
+	std::shared_ptr<Distance> m_distance = NULL;
```

Comment:
No need to set a shared ptr to NULL, the default constructor sets the pointer to nullptr

Reply:
I modified the default constructor to do only SG_ADD. So I guess I'd need to set default values to internal members no?

Reply:
No need for shared ptr, it will by default own a nullptr

Reply:
But primitive types require initialisation yes, otherwise it’s undefined behaviour